packages/webpack-config/src/index.js

```diff
@@ -1,55 +1,76 @@
+const argv = require('yargs').argv;
 const autoprefixer = require('autoprefixer')();
 const CleanWebpackPlugin = require('clean-webpack-plugin');
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 const path = require('path');
 const webpack = require('webpack');
 const dd = require('./dd.js');
 
+let mode = 'production';
+if (argv.mode === 'development' || argv.mode === 'production') {
```

Comment:
We shouldn't rely on the mode flag to derive this. For instance using `.env` or running the command with `NODE_ENV=production webpack` are also valid.

The mode flag is used to preset the environment variables. So its the environment variables that we should use as the source of truth.

**[mode option development](https://webpack.js.org/concepts/mode/)**

>  Provides process.env.NODE_ENV with value development. 

So this line should be something similar to:

`let mode = process.env.NODE_ENV || 'development';`

Reply:
Hey Jenna, I can understand that we should also support the `NODE_ENV` environment variable, but the problem is, setting Webpack's `mode` flag does not modify the `process.env.NODE_ENV` variable, so we need to check both. How about this logic -

If `NODE_ENV` env var is set, use this as the mode.
Else if the `mode` argument is set, use this as the mode.
Else if the `WEBPACK_SERVE` env var is set, set the mode to development.

Reply:
Oh really :(

Yeah if thats the case then I think I like your logic. It covers all bases.